public/systems_monitoring_graphs.js

```diff
@@ -0,0 +1,1087 @@
+                .attr("height", Height + Margin.top + Margin.bottom)
+                .append("g")
+                .attr("transform", "translate(" + Margin.left + "," + Margin.top + ")");
+
+            // Add X axis
+            let x = d3.scaleTime().domain(d3.extent(data, d => new Date(d.datetime))).range([1, Width]);
+            let xAxis = svg.append("g")
+                .attr("transform", "translate(0," + Height + ")")
+                .call(d3.axisBottom(x).ticks(d3.timeDay.every(1)).tickFormat(dayTimeFormat));
+            
+            // Add the X gridlines
+            svg.append("g")			
+                .attr("class", "diskXGrid")
+                .attr("transform", "translate(0," + Height + ")")
+                .call(d3.axisBottom(x)
+                    .tickSize(-Height)
+                    .tickFormat("")
+                )
+            
+            // Rotate X axis ticks
+            xAxis.selectAll("text")
+                .style("text-anchor", "end")
+                .attr("dx", "-.8em")
+                .attr("dy", ".15em")
+                .attr("transform", "rotate(-65)");
+
+            // Add X axis label
+            svg.append("text")
+                .attr("text-anchor", "end")
+                .attr(
+                    "transform",
+                    "translate(" + Width / 2 + " ," + (Height + Margin.top + 50) + ")"
+                )
+                .style("text-anchor", "middle")
+                .text("Date (dd:hh:m)");
+
+            // Add Y axis label
+            svg.append("text")
+                .attr("transform", "rotate(-90)")
+                .attr("y", 0 - Margin.left)
+                .attr("x", 0 - Height / 2)
+                .attr("dy", "1em")
+                .style("text-anchor", "middle")
+                .text("Disc Usage (GB)");
+                
+            let yLimit = data[0].disk_usage.total
+            // Add Y axis
+            let decimalFormatter = d3.format(".2s");
+            let y = d3.scaleLinear()
+                // Add 100 to yLimit to increase brushing area
+                .domain([0, yLimit + 100])
+                .range([ Height, 0 ]);
+                svg.append("g")
+                .call(d3.axisLeft(y).ticks(5).tickFormat((d) => decimalFormatter(d).replace('G', 'GB')))      
+
+            // Add the Y gridlines
+            svg.append("g")			
+                .attr("class", "diskGrid")
+                .call(d3.axisLeft(y)
+                    .tickSize(-Width)
+                    .tickFormat("")
+                )
+
+            // Get time when the system stopped & when it was restarted
+            let stop = [], start = [], currentTime = new Date();
+            data.forEach((d, i, n)=> {
+                if (i == (n.length-1)) {
+                    let difference_ms = (currentTime.getTime() - d.datetime.getTime()) / 60000,
+                    difference_minutes = Math.floor(difference_ms % 60);
+                    if (difference_minutes > 10) {
```

Comment:
Assuming the minutes calculation is now correct - I think the watcher is now sleeping for 10 minutes between runs (check with the KK team if unsure) - so we should wait longer before reporting an outage e.g. 30 minutes.

Reply:
Updated thanks